snuba/query/query_processor.py

```diff
@@ -1,27 +1,29 @@
 from abc import ABC, abstractmethod
-from typing import Any, Generic, Mapping, TypeVar
+from typing import Any, Mapping
 
 from snuba.query.query import Query
 from snuba.request.request_settings import RequestSettings
 
-ExtensionData = Mapping[str, Any]
-
-TQueryProcessContext = TypeVar("TQueryProcessContext")
 
-
-class QueryProcessor(ABC, Generic[TQueryProcessContext]):
+class QueryProcessor(ABC):
```

Comment:
After spending some time trying to figure out what was the right abstraction on top of query processors and extensions processors I came to the conclusions that there is no reason for a common interface:
- they require different parameters in the only method they have
- they are never used interchangeably and should never be used interchangeably
- they serve two different purposes in the process.

Reply:
I think this makes sense. I think you could abstract it down to "query processors update the query and don't have parameters" and "query extensions update the query and have parameters" but that seems like more abstraction than is necessary here, especially if these two things evolve in different directions. Keeping them separate makes sense to me.